lib/helpers/deep-extend.js

```diff
@@ -1,12 +1,16 @@
 "use strict";
 
+function isObject(v) {
+	return v != null && v.constructor.name === "Object";
```

Comment:
It can throw error if `v` is created as `Object.create(null)`
```
TypeError: Cannot read property 'name' of undefined
```

Reply:
Do you have any code suggestion?
I checked the `lodash` source code for its `isPlainObject` method, but it has lot of codes

Reply:
Yeah, I've also checked, it's very big and I think that's very slow.
What was the exact problem with the original code? Maybe better if we extend it

Reply:
> and I think that's very slow.

not a big problem, because It will only run once at compile

Reply:
In this case, just check that there is `v.constructor`. If yes, check the name, if not check the typeof.

Reply:
what about this?
```js
function isObjectHasKey(v) {
	if (v == null || typeof v !== "object") return false;
	return Object.keys(v).length > 0;
}
```